pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -66,6 +66,10 @@ def ensure_{{name}}(object arr, copy=True):
             return arr
         else:
             return arr.astype(np.{{dtype}}, copy=copy)
+{{if na_val == "nan"}}
```

Comment:
hmm i did request this, but agree the performance tradeoff might not be worthit.

Reply:
i think maybe making an ensure_*_with_na is prob better here (and make it the caller responsible if we have an EA)

Reply:
I went with @jbrockmendel's original `isinstance` check instead for now. For the `ensure_foo_with_na` feature, probably should be another issue to discuss because it would be potentially useful anywhere `ensure_foo` is called.